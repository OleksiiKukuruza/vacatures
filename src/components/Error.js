import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

export const StyledError = styled.div`
  height: 100%;
  width: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  color: ${props => props.theme.error}
`;

const Error = ({ error }) => (
  <StyledError>
    Oops, something went wrong
    <div>
      {error}
    </div>
  </StyledError>
);

Error.propTypes = {
  error: PropTypes.string.isRequired
};

export default Error;
