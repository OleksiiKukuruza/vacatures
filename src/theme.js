const theme = {
  primary: {
    main: '#54b6c8',
    contrastText: '#fff'
  },
  background: '#f1f4f5',
  error: '#f44336',
  spacingUnit: 8
};

export default theme;
