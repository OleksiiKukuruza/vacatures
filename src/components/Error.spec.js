import React from 'react';
import { shallow } from 'enzyme';
import theme from '../theme';
import Error, { StyledError } from './Error';

const setup = (Component, props) => {
  const wrapper = shallow(<Component {...props} />);

  return {
    wrapper
  };
};

describe('Error', () => {
  it('renders Error without crashing', () => {
    const { wrapper } = setup(Error, { error: 'Failure' });
    expect(wrapper).toMatchSnapshot();
  });

  it('renders StyledError without crashing', () => {
    const { wrapper } = setup(StyledError, { theme });
    expect(wrapper).toMatchSnapshot();
  });
});
