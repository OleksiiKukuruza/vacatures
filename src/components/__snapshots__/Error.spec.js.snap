// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error renders Error without crashing 1`] = `
<styled.div>
  Oops, something went wrong
  <div>
    Failure
  </div>
</styled.div>
`;

exports[`Error renders StyledError without crashing 1`] = `
.c0 {
  height: 100%;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  color: #f44336;
}

<div
  className="c0"
/>
`;
